storm-server/src/test/java/org/apache/storm/localizer/AsyncLocalizerTest.java

```diff
@@ -94,7 +93,8 @@
     private ClientBlobStore mockblobstore = mock(ClientBlobStore.class);
 
     private static String getTestLocalizerRoot() {
-        File f = new File("./target/" + Thread.currentThread().getStackTrace()[2].getMethodName() + "/localizer/");
+        File f = new File("./target/" + Thread.currentThread().getStackTrace()[2].getMethodName() + "/localizer/")
+                .getAbsoluteFile();
         f.deleteOnExit();
```

Comment:
I think we should use TmpPath instead of deleteOnExit to clean up test files, as they then get deleted when the test is done rather than when the JVM shuts down. Here's an example https://github.com/apache/storm/blob/23cd491da1ddc3cb684eb0a196101af5af64f6a9/storm-server/src/test/java/org/apache/storm/LocalStateTest.java#L31

Reply:
Thanks for the suggestion, it's better that way.
Will change.

Reply:
Changed all tests that used that `baseDir` to use a `TmpPath`. 

I went a bit further and cleaned the tests a bit. They should be equivalent, but I think that now they are more readable. I hope that's ok.